include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -0,0 +1,378 @@
+//
+// Created by Jan Drewniok on 06.02.23.
+//
+
+#ifndef FICTION_CRITICAL_TEMPERATURE_HPP
+#define FICTION_CRITICAL_TEMPERATURE_HPP
+
+#include "calculating_energy_and_state_type.hpp"
+#include "fiction/algorithms/simulation/sidb/energy_distribution.hpp"
+#include "fiction/algorithms/simulation/sidb/exhaustive_ground_state_simulation.hpp"
+#include "fiction/algorithms/simulation/sidb/occupation_probability_excited_states.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/technology/sidb_charge_state.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+#include "fiction/utils/hash.hpp"
+#include "fiction/utils/math_utils.hpp"
+#include "fiction/utils/truth_table_utils.hpp"
+
+#include <fmt/format.h>
+#include <kitty/bit_operations.hpp>
+#include <kitty/dynamic_truth_table.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <map>
+#include <ostream>
+#include <utility>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * This struct stores the parameters for the `critical_temperature` algorithm.
+ */
+struct critical_temperature_params
+{
+    /**
+     * Gate based simulation is conducted.
+     */
+    bool gate_based_simulation = true;
```

Comment:
If we have such toggles that switch between flavors of the algorithm, we usually use an `enum class` like you can find in `fanout_substitution.hpp` or `graph_coloring.hpp` for instance.